src/arrayMethodSort.js

```diff
@@ -4,8 +4,25 @@
  * Implement method Sort
  */
 function applyCustomSort() {
-  [].__proto__.sort2 = function(compareFunction) {
-    // write code here
+  [].__proto__.sort2 = function (compareFunction) {
+    const defaultCompare = (a, b) => {
+      const sA = String(a);
+      const sB = String(b);
+
+      return (sA > sB) - (sA < sB);
+    };
+
+    const compare = compareFunction || defaultCompare;
+
+    for (let i = 1; i < this.length; i++) {
+      for (let j = 0; j < i; j++) {
+        if (compare(this[i], this[j]) < 0) {
+          [this[i], this[j]] = [this[j], this[i]];
+        }
+      }
+    }
```

Comment:
The bubble sort algorithm used here is correct, but it's worth noting that bubble sort has a worst-case and average complexity of O(n^2), where n is the number of items being sorted. This makes it inefficient on large lists. Consider researching more efficient sorting algorithms like quicksort or mergesort for better performance on large arrays.